src/Tools/dotnet-dsrouter/ADBTcpRouterFactory.cs

```diff
@@ -147,13 +151,41 @@ public override void Start()
             // Enable port reverse.
             _ownsPortReverse = ADBCommandExec.AdbAddPortReverse(_port, Logger);
 
+            _portReverseTaskCancelToken = new CancellationTokenSource();
+            _portReverseTask = Task.Run(() => {
+                while (!_portReverseTaskCancelToken.Token.IsCancellationRequested)
```

Comment:
Good catch! Doing the task async and use await is way to go. I guess we could just make task async and use await Task.Delay(5000, _portReverseTaskCancelToken.Token).ConfigureAwait(false); instead of moving over to .net6+ WaitForNextTickAsync API. But since we moved diagnostics up to .net6 I guess its fine to start taking dependencies to .net6 API's as well, I will move implementation over to above suggestion in both locations.